frontend/src/metabase/query_builder/components/filters/pickers/RelativeDatePicker.tsx

```diff
@@ -0,0 +1,226 @@
+            key={period}
+            primaryColor={primaryColor}
+            selected={operator && unit === period.toLowerCase()}
+            onClick={() => {
+              onCommit([operator, field, intervals, period]);
+            }}
+          >
+            {formatBucketing(period, 1)}
+          </CurrentButton>
+        </TippyPopover>
+      ))}
+    </CurrentContainer>
+  );
+}
+
+export const DATE_PERIODS = ["day", "week", "month", "quarter", "year"];
+
+const TIME_PERIODS = ["minute", "hour"];
+
+// define ALL_PERIODS in increasing order of duration
+const ALL_PERIODS = TIME_PERIODS.concat(DATE_PERIODS);
+
+const getCurrentString = (filter: Filter) =>
+  t`Include ${getCurrentIntervalName(filter)}`;
+
+const CURRENT_INTERVAL_NAME = {
+  day: t`today`,
+  week: t`this week`,
+  month: t`this month`,
+  year: t`this year`,
+  minute: t`this minute`,
+  hour: t`this hour`,
+};
+
+function getCurrentIntervalName(filter: Filter) {
+  if (filter[0] === "time-interval") {
+    return CURRENT_INTERVAL_NAME[
+      filter[3] as keyof typeof CURRENT_INTERVAL_NAME
+    ];
+  }
+  return null;
+}
+
+type Props = {
+  className?: string;
+  filter: Filter;
+  onFilterChange: (filter: any[]) => void;
+  formatter: (value: number) => number;
+  primaryColor?: string;
+};
+
+type State = {
+  showStartingFrom: boolean;
+  showOptions: boolean;
+};
+
+class DatePicker extends Component<Props, State> {
+  state = {
+    showStartingFrom: false,
+    showOptions: false,
+  };
+
+  static defaultProps = {
+    formatter: (value: string) => value,
+  };
+
+  render() {
+    const {
+      filter,
+      onFilterChange,
+      formatter,
+      className,
+      primaryColor,
+    } = this.props;
+    const { showStartingFrom, showOptions } = this.state;
+    const intervals = filter[2];
+    const unit = filter[3];
+    const options = filter[4] || {};
+    const includeCurrent = !!options["include-current"];
+
+    const optionsContent = (
+      <OptionsContainer>
+        {/* <OptionButton icon="arrow_left_to_line">
+          {t`Starting from...`}
+        </OptionButton> */}
+        <OptionButton
+          selected={includeCurrent}
+          primaryColor={primaryColor}
+          icon={includeCurrent ? "check" : "calendar"}
+          onClick={() => {
+            onFilterChange(
+              assoc(filter, 4, {
+                ...options,
+                "include-current": !includeCurrent,
+              }),
+            );
+            this.setState({ showOptions: false });
+          }}
+        >
+          {getCurrentString(filter)}
+        </OptionButton>
+      </OptionsContainer>
+    );
+    return (
+      <div className={cx(className, "flex align-center")}>
+        <NumericInput
+          className="mr2 input border-purple text-right"
+          style={{
+            width: 65,
+            // needed to match Select's AdminSelect classes :-/
+            fontSize: 14,
+            fontWeight: 700,
+            padding: 8,
```

Comment:
Can you do this with `emotion`?

Reply:
Unfortunately it's not working with the layered input component